### PR TITLE
fix: progress plugin custom prefix not work

### PR DESCRIPTION
**Pull request by liruifengv**
## Summary

Fix progress plugin custom prefix not work.

[The docs](https://rsbuild.dev/zh/config/dev/progress-bar) says can modify the progress bar id by setting, but it's not working.

```
export default {
  dev: {
    progressBar: {
      id: 'Some Text',
    },
  },
};
```

Use `progressBar.id` as prefix if it exist.

### before
image

### after
image

## Related Links

- docs: https://rsbuild.dev/zh/config/dev/progress-bar

## Checklist

- [ ] Tests updated.
- [ ] Documentation updated.

### CODE REVIEW

File: packages/core/src/provider/plugins/progress.ts [hunk 1]
```diff
@@ -15,11 +15,12 @@ export const pluginProgress = (): RsbuildPlugin => ({
         return;
       }
 
-      const { ProgressPlugin } = await import('@rspack/core');
+      const prefix = options !== true ? options.id : TARGET_ID_MAP[target];
```
Comment (chenjiahan) [inline packages/core/src/provider/plugins/progress.ts:18]: Thanks for your contribution ❤️

Is `options.id` can be undefined here?

Comment (liruifengv) [reply packages/core/src/provider/plugins/progress.ts:18]: undefined will cause no prefix.

image

Comment (liruifengv) [reply packages/core/src/provider/plugins/progress.ts:18]: See rspack

image

Comment (chenjiahan) [reply packages/core/src/provider/plugins/progress.ts:18]: How about do it as follows, which is more intuitive:

- progressBar.id: undefined -> display the default id
- progressBar.id: '' (empty string) -> display no id

Comment (liruifengv) [reply packages/core/src/provider/plugins/progress.ts:18]: Great suggestion:

```ts
const prefix = (options !== true && options.id !== undefined) ? options.id : TARGET_ID_MAP[target];
```

**Comment by liruifengv** [comment]
An unrelated point, your GitHub profile's Twitter link was 404. @chenjiahan

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
LGTM 😄

**Comment by chenjiahan** [comment]
> your GitHub profile's Twitter link was 404

Thanks...I have updated the link 😂

**Pull request merged**